src/Keboola/StorageApi/Client.php

```diff
@@ -147,6 +158,22 @@ private function initClient()
         ]);
     }
 
+    /**
+     * @param $maxJobPollWaitPeriodSeconds
+     * @return callable
+     */
+    private static function getDefaultJobPollDelay($maxJobPollWaitPeriodSeconds)
+    {
+        return function ($tries) use ($maxJobPollWaitPeriodSeconds) {
+            return min(pow(2, $tries), $maxJobPollWaitPeriodSeconds);
+        };
+    }
+
+    public function setJobPollDelayMethoc(callable $jobPollDelay)
+    {
+        $this->retryDelay = $jobPollDelay;
```

Comment:
I am not sure about that setter, client should be immutable.

Reply:
that's right, that was originally my idea :-/ how would you solve this situation?

we need an instance, that will check for a user/project feature and then we can finally decide which delay function to use? would you create and return new client?

Reply:
and there's a typo too: `setJobPollDelayMethoc` => `setJobPollDelayMethod`

Reply:
doh

Reply:
yes I think creation of new client will be ok. It will no affect performance, api calls count will be same. But i am not sure where it will be used, but everything like URL, runId must be set for both clients...

Reply:
There is still mutable `setRunId` method but i think it should be also moved to constructor only in some major release.

Reply:
ok, I'll rip out the setter.